### DISCUSSION TITLE
Teleport losing all the data

### DISCUSSION THREAD

**Discussion by Aman1994** [gravitational/teleport#10246] [Q&A] [answered]
Hi,
I have deployed the `teleport-cluster` helm chart in Kubernetes. After creating the users and token , when i delete my kube pod and a new pod is created automatically all the data are not lost even though it has a PVC.

Can you please let me know why its happening ?

**Comment by stevenGravy** [comment]
Can you share the k8s env you're using?

**Reply by KlavsKlavsen** [reply]
@stevenGravy its Kubernetes 1.22 currently - installed with kubeadm on physical hardware. Is that what you were looking for?
All other services we run on the cluster works fine.. and the pod has its PVC still.. it simply forgets these things appearently upon pod kill (and start of new pod). We can find nothing of relevance in log output when new pod starts :(

What can we give you to help us debug? (we'll share helm values used and our teleport.yaml config file)

**Comment by KlavsKlavsen** [comment, marked as answer]
Sorry guys - I just looked through the pod and the issue is that the chart ONLY actually mounts the data pvc IF in standalone mode. So Custom mode breaks persistency. We'll adjust chart (for now to add support for standalone mode to have externally delivered configmap and submit PR once its confirmed to work

**Reply by webvictim** [reply]
Yes, that's correct. I suspect that the better PR would actually be to separate out the PVC config into a separate `storage` section so that it can be enabled when `custom` mode is used.
